### PR TITLE
[#12] Add SSM Parameters (from Parameter Store) for encrypted variables to be accessed inside ECS container environment

### PR DESCRIPTION
**Pull request by liamstevens111**
- #12 

**Builds from Elasticache PR: https://github.com/liamstevens111/terraform-infrastructure-example/pull/29**

## What happened 👀

- [x] Add SSM Parameters for `3` sensitives variables to be used inside the application - RDS+Elasticache connection, and secret key
- [x] Added TaskExecutionRole for ECS to fetch these values (no KMSDecrypt permission is needed I think because using default AWS KMS key)
- [x] Moved the `3` variables from `environment` to `secrets` inside `container definitions` for ECS to access the Parameter Store

## Insight 📝

Refactor same time as when doing project (ie not fetching inline in container) or switch to Secrets Manager if required.

**Question: Is the preferred way of creating these ssm parameters still via environment variables inside terraform? And what's difference between terraform sensitive variables, they're not encrypted inside the container at runtime? Or because more control over encryption?**

## Proof Of Work 📹

Showing Parameter Store variables with type `SecureString`:

Screenshot 2022-12-21 at 15 01 21

Value showing the RDS connection used inside Elixir / Phoenix inside the variable:
`Blanked password due to public repo although the RDS instance isn't publicly accessible`

Screenshot 2022-12-21 at 15 58 29

CloudWatch logs showing it can correctly access the variables inside the container (DB connects):
(Initially had problem because env variable case sensitive on linux, had to make uppercase for SSM)
Screenshot 2022-12-21 at 16 06 42

Working code with all above at https://github.com/liamstevens111/terraform-elixir-repo/tree/develop:
Shows elasticache redis `GET` and `SET` (redix library), s3 static asset, and redis address from SSM parameter store (not tf cloud) and rds user auth:

Screenshot 2022-12-22 at 13 52 17

### CODE REVIEW

File: modules/ssm/main.tf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+resource "aws_ssm_parameter" "main" {
+  for_each = var.parameters
```
Comment (andyduong1920) [inline modules/ssm/main.tf:2]:
```suggestion
  for_each = var.parameters
  
```

**Review by andyduong1920 — APPROVED** [review]

File: modules/ssm/main.tf (hunk 1 above)
Comment (liamstevens111) [reply modules/ssm/main.tf:2]: @andyduong1920 Nice! Here [363bc48](https://github.com/liamstevens111/terraform-infrastructure-example/pull/30/commits/363bc48edc431642aecd4a5ee21b04bf5a348e77)

**Review by hoangmirs — APPROVED** [review]

**Comment by liamstevens111** [comment]
Closed and incorporated into refactor repo here: https://github.com/liamstevens111/terraform-aws-infrastructure

**Comment by liamstevens111** [comment]
Closed and incorporated into refactor repo here: https://github.com/liamstevens111/terraform-aws-infrastructure

**Pull request merged**